code/datums/outfits/_outfit.dm

```diff
@@ -0,0 +1,319 @@
+	.["shoes"] = shoes
+	.["head"] = head
+	.["mask"] = mask
+	.["ears"] = ears
+	.["glasses"] = glasses
+	.["id"] = id
+	.["l_pocket"] = l_pocket
+	.["r_pocket"] = r_pocket
+	.["suit_store"] = suit_store
+	.["r_hand"] = r_hand
+	.["l_hand"] = l_hand
+	.["slot_in_back"] = backpack_contents
+	.["slot_in_belt"] = belt_contents
+	.["slot_in_boot"] = shoe_contents
+	.["slot_in_suit"] = suit_contents
+	.["slot_in_accessory"] = webbing_contents
+	.["slot_in_head"] = head_contents
+	.["slot_in_r_store"] = r_pocket_contents
+	.["slot_in_l_store"] = l_pocket_contents
+	.["implants"] = implants
+
+/// Copy most vars from another outfit to this one
+/datum/outfit/proc/copy_from(datum/outfit/target)
+	name = target.name
+	w_uniform = target.w_uniform
+	wear_suit = target.wear_suit
+	back = target.back
+	belt = target.belt
+	gloves = target.gloves
+	shoes = target.shoes
+	head = target.head
+	mask = target.mask
+	ears = target.ears
+	glasses = target.glasses
+	id = target.id
+	l_pocket = target.l_pocket
+	r_pocket = target.r_pocket
+	suit_store = target.suit_store
+	r_hand = target.r_hand
+	l_hand = target.l_hand
+	backpack_contents = target.backpack_contents
+	belt_contents = target.belt_contents
+	shoe_contents = target.shoe_contents
+	suit_contents = target.suit_contents
+	webbing_contents = target.webbing_contents
+	head_contents = target.head_contents
+	r_pocket_contents = target.r_pocket_contents
+	l_pocket_contents = target.l_pocket_contents
+	implants = target.implants
+	return TRUE
+
+/datum/outfit/proc/save_to_file()
+	var/stored_data = get_json_data()
+	var/json = json_encode(stored_data)
+	//Kinda annoying but as far as i can tell you need to make actual file.
+	var/f = file("data/TempOutfitUpload")
+	fdel(f)
+	WRITE_FILE(f, json)
+	usr << ftp(f, "[name].json")
+
+
+/datum/outfit/proc/load_from(list/outfit_data)
+	name = outfit_data["name"]
+	w_uniform = text2path(outfit_data["w_uniform"])
+	wear_suit = text2path(outfit_data["wear_suit"])
+	back = text2path(outfit_data["back"])
+	belt = text2path(outfit_data["belt"])
+	gloves = text2path(outfit_data["gloves"])
+	shoes = text2path(outfit_data["shoes"])
+	head = text2path(outfit_data["head"])
+	mask = text2path(outfit_data["mask"])
+	ears = text2path(outfit_data["ears"])
+	glasses = text2path(outfit_data["glasses"])
+	id = text2path(outfit_data["id"])
+	l_pocket = text2path(outfit_data["l_pocket"])
+	r_pocket = text2path(outfit_data["r_pocket"])
+	suit_store = text2path(outfit_data["suit_store"])
+	r_hand = text2path(outfit_data["r_hand"])
+	l_hand = text2path(outfit_data["l_hand"])
+	var/list/backpack = outfit_data["backpack_contents"]
+	backpack_contents = list()
+	for(var/item in backpack)
+		var/itype = text2path(item)
+		if(itype)
+			backpack_contents[itype] = backpack[item]
+	var/list/impl = outfit_data["implants"]
+	implants = list()
+	for(var/I in impl)
+		var/imptype = text2path(I)
+		if(imptype)
+			implants += imptype
+	return TRUE
+
+/datum/outfit/proc/from_mob(mob/living/base)
```

Comment:
Should this be handling container contents at all? Also could use autodocs on a bunch of these.

Reply:
This is not my code, I just moved this from another file. I don't even know if the file things even work